frontend/catalyst/api_extensions/quantum_operators.py

```diff
@@ -315,18 +314,30 @@ class MidCircuitMeasure(HybridOp):
 
     binder = qmeasure_p.bind
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.postselect = self.in_classical_tracers[-2]
-        self.reset = self.in_classical_tracers[-1]
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        in_classical_tracers,
+        out_classical_tracers,
+        regions: List[HybridOpRegion],
+        reset: bool = None,
+        postselect: int = None,
+    ):
+        HybridOp.__init__(self, in_classical_tracers, out_classical_tracers, regions)
+        self.reset = reset
+        self.postselect = postselect
+        # self.mv = self.out_classical_tracers[0]
 
-    def trace_quantum(self, ctx, device, trace, qrp) -> QRegPromise:
-        op = self
-        wire = op.in_classical_tracers[0]
+    # pylint: disable=too-many-arguments
+    def trace_quantum(self, ctx, device, trace, qrp, postselect_mode=None) -> QRegPromise:
+        wire = self.in_classical_tracers[0]
         qubit = qrp.extract([wire])[0]
-        # qubit2 = op.bind_overwrite_classical_tracers(ctx, trace, qubit)
-        # TODO: execute post-selection depending on qnode config
-        qubit2 = op.bind_overwrite_classical_tracers(ctx, trace, qubit, postselect=op.postselect)
+        if postselect_mode != "hw-like":
```

Comment:
Nitpick, I prefer `==` conditions when possible

Reply:
Yes especially in this case I think applying the post-select should be the default :)